WordPress/src/main/java/org/wordpress/android/ui/stats/StatsPublishedPostsFragment.java

```diff
@@ -0,0 +1,173 @@
+        if (!hasPublishedPostsAndPages()) {
+            return null;
+        }
+        return ((PublishedPostsAndPagesModel) mDatamodels[0]).getPublishedPostsAndPages();
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
+        View view = super.onCreateView(inflater, container, savedInstanceState);
+
+        // No labels on columns for this module
+        TextView entryLabel = (TextView) view.findViewById(R.id.stats_list_entry_label);
+        entryLabel.setVisibility(View.GONE);
+        TextView totalsLabel = (TextView) view.findViewById(R.id.stats_list_totals_label);
+        totalsLabel.setVisibility(View.GONE);
+        return view;
+    }
+
+    /**
+     * This module has a slightly different UI on the top of it.
+     *
+     * Changes are:
+     * - "no-labels" on columns.
+     * - Title is created at run-time by using the number of posts published in the timeframe.
+     */
+    @Override
+    protected void showHideNoResultsUI(boolean showNoResultsUI) {
+        super.showHideNoResultsUI(showNoResultsUI);
+        if (showNoResultsUI) {
+            // rewrite here the empty label, since it doesn't include bold text and it's 1 line only.
+            String label = "<b>" + getString(R.string.stats_empty_published_posts_desc) + "</b>";
+            mEmptyLabel.setText(Html.fromHtml(label));
+            mEmptyLabel.setVisibility(View.VISIBLE);
+            mModuleTitleTextView.setText(R.string.stats_empty_published_posts_title);
+        } else {
+            mModuleTitleTextView.setText(getTitle());
+        }
+    }
+
+    @Override
+    protected boolean isViewAllOptionAvailable() {
+        return hasPublishedPostsAndPages() && getTopPostsAndPages().size() > MAX_NUM_OF_ITEMS_DISPLAYED_IN_LIST;
+    }
+
+    @Override
+    protected boolean isExpandableList() {
+        return false;
+    }
+
+    @Override
+    protected int getEntryLabelResId() {
+        return R.string.stats_entry_posts_and_pages;
+    }
+
+    @Override
+    protected int getTotalsLabelResId() {
+        return R.string.stats_totals_views;
+    }
+
+    @Override
+    protected int getEmptyLabelTitleResId() {
+        return R.string.stats_empty_published_posts_title;
+    }
+
+    @Override
+    protected int getEmptyLabelDescResId() {
+        return NO_STRING_ID;
+    }
+
+    @Override
+    protected StatsService.StatsEndpointsEnum[] getSectionsToUpdate() {
+        return new StatsService.StatsEndpointsEnum[]{
+                StatsService.StatsEndpointsEnum.PUBLISHED_POSTS
+        };
+    }
+
+    @Override
+    public String getTitle() {
+        return getString(R.string.stats_view_published);
+    }
+
+
+    private class PublishedPostsAndPagesAdapter extends ArrayAdapter<PostModel> {
+
+        private List<PostModel> list;
+        private final LayoutInflater inflater;
+        private final int localTableBlogID;
```

Comment:
This field is unused. Will it be needed in the future or is it worth taking out?